pir/cpp/client.cpp

```diff
@@ -41,69 +46,69 @@ StatusOr<std::unique_ptr<PIRClient>> PIRClient::Create(
 }
 
 StatusOr<PIRPayload> PIRClient::CreateRequest(std::size_t index) const {
-  if (index >= this->DBSize()) {
+  const auto poly_modulus_degree =
+      context_->Parameters()->GetEncryptionParams().poly_modulus_degree();
+  if (index >= DBSize()) {
     return InvalidArgumentError("invalid index");
   }
-  std::vector<std::int64_t> request(DBSize(), 0);
-  request[index] = 1;
-
-  return this->encryptRequestBuffer(request);
-}
+  if (index >= poly_modulus_degree) {
+    // Not yet implemented
+    return InvalidArgumentError("More than 1 CT needed for selection vector");
```

Comment:
Please add an issue to the repo for this too.

Reply:
Done. https://github.com/OpenMined/PIR/issues/9